### ISSUE TITLE
[@tiptap/suggestion] update method of the view() in Plugin gets called 3 times

### ISSUE THREAD

**Issue by samkim130** [ueberdosis/tiptap#3851]
### What’s the bug you are facing?

See 
https://github.com/ueberdosis/tiptap/blob/develop/packages/suggestion/src/suggestion.ts#L70
After console logging, we noticed it's getting called 3 times for every key press.

It causes an issue when the cursor is in the middle of the text, it calls `onStart` and `onExit` as well on the other two times.

(also might be ProseMirror or some root src package issue, but not sure where else to file this issue)

### Which browser was this experienced in? Are any special extensions installed?

Any browser, no special extensions

### How can we reproduce the bug on our side?

Use Mentions
1. Got to root https://github.com/ueberdosis/tiptap/blob/develop/packages/suggestion/src/suggestion.ts#L70 file in node modules
2. just set some value `let x = 0;` outside the scope in the file
3. add `console.log(x++);` at the beginning of the [update](https://github.com/ueberdosis/tiptap/blob/develop/packages/suggestion/src/suggestion.ts#L70) function
4. see console output as you type

For the bug behavior:
1. go to middle of text with suggestion and mention enabled
2. try tagging with @

### Can you provide a CodeSandbox?

https://codesandbox.io/s/elegant-antonelli-kojzk0?file=/src/App.vue

Basically copy of https://github.com/ueberdosis/tiptap/tree/main/demos/src/Nodes/Mention/Vue

but try console.logging suggestion src file

### What did you expect to happen?

only call update once, so that the @ mentioning feature works mid text

### Anything to add? (optional)

if you give me some pointers on where to look, i'll also try to solve this
I'm just not sure what even calls this `update` method

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by lesliewu1** [comment]
Bumping this as it is still happening with no fix :(
